src/bellmanford.jl

```diff
@@ -105,3 +108,28 @@ function has_negative_edge_cycle{V, D}(
     edge_inspector = VectorEdgePropertyInspector{D}(edge_dists)
     has_negative_edge_cycle(graph, edge_inspector)
 end
+
+function enumerate_paths{V,D<:Number}(state::BellmanFordStates{V,D}, dest::Vector{V})
+    parents = state.parents
+    hasparent = state.hasparent
+    
+    num_dest = length(dest)
+    all_paths = Array(Vector{V},num_dest)
+    for i=1:num_dest
+        all_paths[i] = V[]
+        index = dest[i]
+        if hasparent[index] || parents[index] == index
+            while hasparent[index]
```

Comment:
It looks like everywhere you're using `extractRoute` you're running Dijkstra's algorithm on the full graph and then throwing it away - if that's so, you should definitely be using A* (and the obvious heuristic in both cases is "distance as the crow flies"). In general, if you don't know the heuristic to use then you should use BFS, which is equivalent to A* with the heuristic f(x) = 0 for all x.

> My understanding is that in huge graphs, you seldom ever actually want to store the paths to all vertices (in memory) anyway, esp. when you already have parents.

Ah, I thought @sbromberger needed this for his centrality work, but I guess I misunderstood.

> since you already have parents, which is the vector of common prefixes that you're talking about.

Yes! But all that hard work is thrown away :-) Suppose you care about the shortest paths from node 1 to nodes 6 and 7, and the shortest paths go 
1 -> 2 -> 3 -> 4 -> 5 -> 6 
1 -> 2 -> 3 -> 4 -> 5 -> 7
then this code will calculate the 1 -> 2 -> 3 -> 4 -> 5 part twice.

On the other hand:
 - `dijkstra_shortest_paths_explicit` worked the same way,
 - your ambulance-routing example is compelling,
so I withdraw this objection (though it would be really nice if we had some realistic examples to benchmark on so we could figure out if it was worth memoizing paths).